### PR TITLE
Support installing the agent in a custom namespace

### PR DESCRIPTION
**Pull request by asankov-cb**
This change makes the operator configurable about the namespace in which the agent will be installed.

The change is also propagated to the Helm charts.

### CODE REVIEW

File: charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml [hunk 1]
```diff
@@ -4,13 +4,13 @@ metadata:
   labels:
     control-plane: operator
     octarine: ignore
-  name: cbcontainers-dataplane
+  name: {{ default "cbcontainers-dataplane" .Values.namespace }}
```
Comment (ltsonov-cb) [inline charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml:7]: How does this play with helm's `--namespace` and `--create-namespace` flags? 

When I was testing this; choosing `--namespace X` where X is created by the chart did not work (helm required that the namespace exists BEFORE deploying it)
Using `--create-namespace` created it but failed because then the chart tries to re-deploy it.

I got it working with the previous version by specifying labels that enable helm to "adopt" the existing namespace (see deleted lines in readme.md). However, if we not support this method 100% - I think we should change the installation instructions to use `--create-namespace` always and remove this from the chart. Much simpler. 

(We still need the namespace in the non-chart version though) 

WDYT?

File: charts/cbcontainers-operator/README.md [hunk 2]
```diff
@@ -39,7 +25,7 @@ helm repo update
 helm install cbcontainers-operator TODO-chart-repo/TODO-chart-name --namespace X
```
Comment (ltsonov-cb) [inline charts/cbcontainers-operator/README.md:25]: If we have these values; can we fix that now too ?

**Comment by ltsonov-cb** [comment]
Not sure where the tests for this are stored (Sam should know) - but we need a test for this. I am ok with logging at as a task and handling separately.

File: charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml (hunk 1 above)
Comment (asankov-cb) [reply charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml:7]: I think there might be a use-case where teams want to install the agent/operator in one namespace, but the chart itself in a different one. I think this will break, if we always install the operator along the chart. I think the possible use-cases are:
1. install the chart in a new namespace, install the operator/agent in a new, but different namespace
2. install the chart in a new namespace, install the operator/agent in the same new namespace
3. chart in a new namespace, operator/agent in an existing namespace
4. chart in an existing namespace, operator/agent in a different existing namespace
5. chart in an existing namespace, operator/agent in the same existing namespace
6. chart in an existing namespace, operator/agent in a new namespace

right now,
1) is possible via:
```sh
helm install .... -n new-namespace --create-namespace --set createNamespace=true
```

2) should also be possible:
```sh
helm install .... -n new-namespace --create-namespace --set createNamespace=false --set namespace=new-namespace
```

3) is possible:
```sh
helm install .... -n new-namespace --create-namespace --set createNamespace=false --set namespace=old-namespace
```

4) is possible:
```sh
helm install .... -n old-namespace --set createNamespace=false --set namespace=old-namespace-2
```

5) is possible:
```sh
helm install .... -n old-namespace --set createNamespace=false --set namespace=old-namespace
```

6) is possible:
```sh
helm install .... -n old-namespace --set createNamespace=false --set namespace=new-namespace --set createNamespace=true
```

All in all, I like having 2 less values in the `values.yaml` file, but I don't think we can support all scenarios if we only use the native Helm support for namespace creation

File: charts/cbcontainers-operator/README.md (hunk 2 above)
Comment (asankov-cb) [reply charts/cbcontainers-operator/README.md:25]: yes and no. I think the actual repo we push the chart to is private (need to check for sure). so it's better to leave these as they are (or just say to install from source) rather than link to a private repo. that is, until we can find a public repo for the charts.

File: charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml (hunk 1 above)
Comment (asankov-cb) [reply charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml:7]: NOTE: in the current version we do not have the `createNamespace` flag, but that is coming in the next PR (or I can actually cherry-pick the commit to this one, because it looks important to have it right away)

Comment (ltsonov-cb) [reply charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml:7]: Yes, a flag like `createNamespace` that controls whether the chart tries to deploy a namespace or not is what was missing here. With it; I think all is ok (and docs should mention this decoupled option with maybe 1-2 scenarios from above). 

So please cherry pick it here if it's not a big problem. 

My only suggestion is to think of a name that doesn't look so similar to helm's `--create-namespace` to avoid confusion. Maybe `createNamespaceForOperator` or maybe `useExistingNamespace` (with the opposite flag functionality of course). WDYT?

**Comment by ltsonov-cb** [comment]
LGTM except the comments above

File: charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml (hunk 1 above)
Comment (asankov-cb) [reply charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml:7]: I cherry-picked the `createNamespace` flag.

about the naming: I agree that our flag being the same as the Helm one can be confusing. However, `createNamespaceForOperator` is too wordy for me, and `useExistingNamespace` is more confusing IMO. Let's crowd-source this decision in the team chat :) 

FTR I tried all scenarios in a kind cluster and they work:

## 1. works
```sh
$ k get ns
NAME                 STATUS   AGE
default              Active   71s
kube-node-lease      Active   72s
kube-public          Active   72s
kube-system          Active   72s
local-path-storage   Active   64s

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace new-ns --create-namespace --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=new-ns-2
W1123 10:38:14.671993   67481 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:38:14.804542   67481 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:38:14 2022
NAMESPACE: new-ns
STATUS: deployed
REVISION: 1
TEST SUITE: None

$ k get ns
NAME                 STATUS   AGE
default              Active   2m55s
kube-node-lease      Active   2m56s
kube-public          Active   2m56s
kube-system          Active   2m56s
local-path-storage   Active   2m48s
new-ns               Active   3s
new-ns-2             Active   3s
```

## 2. works
```sh
$ k get ns
NAME                 STATUS   AGE
default              Active   5m32s
kube-node-lease      Active   5m33s
kube-public          Active   5m33s
kube-system          Active   5m33s
local-path-storage   Active   5m25s

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace new-ns --create-namespace --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=new-ns --set createNamespace=false
W1123 10:41:11.400703   67892 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:41:11.518693   67892 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:41:10 2022
NAMESPACE: new-ns
STATUS: deployed
REVISION: 1
TEST SUITE: None

$ k get ns
NAME                 STATUS   AGE
default              Active   5m52s
kube-node-lease      Active   5m53s
kube-public          Active   5m53s
kube-system          Active   5m53s
local-path-storage   Active   5m45s
new-ns               Active   3s
```
## 3. works

```sh
$ k get ns
NAME                 STATUS   AGE
default              Active   7m59s
kube-node-lease      Active   8m
kube-public          Active   8m
kube-system          Active   8m
local-path-storage   Active   7m52s

$ k create ns install-ns
namespace/install-ns created

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace new-ns --create-namespace --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=install-ns --set createNamespace=false
W1123 10:43:50.861702   68404 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:43:50.998526   68404 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:43:50 2022
NAMESPACE: new-ns
STATUS: deployed
REVISION: 1
TEST SUITE: None

$ k get ns
NAME                 STATUS   AGE
default              Active   8m31s
install-ns           Active   28s
kube-node-lease      Active   8m32s
kube-public          Active   8m32s
kube-system          Active   8m32s
local-path-storage   Active   8m24s
new-ns               Active   3s
```

## 4. works

```sh
$ k get ns
NAME                 STATUS   AGE
chart-ns             Active   39s
default              Active   11m
install-ns           Active   18s
kube-node-lease      Active   11m
kube-public          Active   11m
kube-system          Active   11m
local-path-storage   Active   11m

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace chart-ns --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=install-ns --set createNamespace=false
W1123 10:47:10.362255   68946 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:47:10.451524   68946 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:47:09 2022
NAMESPACE: chart-ns
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

## 5. works

```sh
$ k create ns ns-for-both
namespace/ns-for-both created

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace ns-for-both --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=ns-for-both --set createNamespace=false
W1123 10:51:44.221457   70529 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:51:44.309717   70529 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:51:43 2022
NAMESPACE: ns-for-both
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

### 6. works

```sh
$ k create ns chart-ns
namespace/chart-ns created

$ helm install operator ./charts/cbcontainers-operator/cbcontainers-operator-chart --namespace chart-ns --values ./charts/cbcontainers-operator/cbcontainers-operator-chart/values.yaml --set namespace=new-ns --set createNamespace=true
W1123 10:56:36.964351   71262 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W1123 10:56:37.057550   71262 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
NAME: operator
LAST DEPLOYED: Wed Nov 23 10:56:36 2022
NAMESPACE: chart-ns
STATUS: deployed
REVISION: 1
TEST SUITE: None

$ k get ns
NAME                 STATUS   AGE
chart-ns             Active   25s
default              Active   21m
kube-node-lease      Active   21m
kube-public          Active   21m
kube-system          Active   21m
local-path-storage   Active   21m
new-ns               Active   2s
```

**Review by ltsonov-cb — APPROVED** [review]

File: charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml (hunk 1 above)
Comment (ltsonov-cb) [reply charts/cbcontainers-operator/cbcontainers-operator-chart/templates/operator.yaml]: We ended up with `craeteOperatorNamespace` and `operatorNamespace`
